javalib/src/main/scala/java/io/InputStreamReader.scala

```diff
@@ -91,6 +91,8 @@ class InputStreamReader(private[this] var in: InputStream,
         readMoreThroughOutBuf(cbuf, off, len)
       }
     }
+    else
+      -1
```

Comment:
Oh, (l.149) is not ok, could you fix it, it should be:
```scala
if (decoder.flush(out).isOverflow) {
  InputStreamReader.Overflow
} else {
  ...
}
```

and I see that (l.72) has that same problem, it should be:
```scala
if (len == 0) { 
  0
} else if (outBuf.hasRemaining) {
  ...
```